examples/lb3-application/src/__tests__/acceptance/lb3app.acceptance.ts

```diff
@@ -0,0 +1,190 @@
+      const currentDate = new Date();
+      const currentHour = currentDate.getHours();
+      const response = await client.get('/api/CoffeeShops/status').expect(200);
+
+      if (currentHour >= 6 && currentHour < 20) {
+        expect(response.body.status).to.eql('We are open for business.');
+      } else {
+        expect(response.body.status).to.eql(
+          'Sorry, we are closed. Open daily from 6am to 8pm.',
+        );
+      }
+    });
+
+    it('gets external route in application', async () => {
+      await client.get('/ping').expect(200, 'pong');
+    });
+  });
+
+  context('LoopBack 3 authentication', () => {
+    // tslint:disable-next-line:no-any
+    let User: any;
+
+    before(() => {
+      User = lb3App.models.User;
+    });
+
+    it('creates and logs in a User', async () => {
+      const user = await User.create({
+        email: 'created@email.com',
+        password: 'L00pBack!',
+      });
+
+      expect(user.email).to.eql('created@email.com');
+
+      await User.login({
+        email: 'created@email.com',
+        password: 'L00pBack!',
+      });
+    });
+
+    it('makes an authenticated request', async () => {
+      await User.create({
+        email: 'sample@email.com',
+        password: 'L00pBack!',
+      });
+
+      // await client.get('/api/CoffeeShops/greet').expect(401);
+
+      const token = await User.login({
+        email: 'sample@email.com',
+        password: 'L00pBack!',
+      });
+
+      const response = await client
+        .get(`/api/CoffeeShops/greet?access_token=${token.id}`)
+        .expect(200);
+
+      expect(response.body.undefined).to.eql('Hello from this Coffee Shop');
+    });
+
+    it.skip("denies request made by another user's access token", async () => {
+      const users = await User.create([
+        {
+          email: 'sample@email.com',
+          password: 'L00pBack!',
+        },
+        {
+          email: 'other@email.com',
+          password: 'L00pBack!',
+        },
+      ]);
+
+      const token = await User.login({
+        email: users[0].email,
+        password: 'L00pBack!',
+      });
+
+      const otherToken = await User.login({
+        email: users[1].email,
+        password: 'L00pBack!',
+      });
+
+      const response = await client
+        .get(`/api/Users/${users[0].id}?access_token=${token.id}`)
+        .expect(200);
+
+      expect(response.body).to.containEql({
+        email: users[0].email,
+        id: users[0].id,
+      });
+
+      await client
+        .get(`/api/Users/${users[0].id}?access_token=${otherToken.id}`)
+        .expect(401);
+    });
+  });
+
+  context('OpenAPI spec', () => {
+    let apiSpec: OpenApiSpec;
+    let swaggerSpec: OpenApiSpec;
+    let lb3Spec: OpenApiSpec;
+
+    before(async () => {
+      apiSpec = app.restServer.getApiSpec();
+      swaggerSpec = await generateSwaggerSpec(lb3App);
+      const result = await swagger2openapi.convertObj(swaggerSpec, {});
```

Comment:
I find this approach problematic. You are basically repeating the same steps used by LB3 booter to convert from Swagger to OpenAPI. If these steps are flawed and produce incorrect spec, then your test will not catch that.

A better solution is to explicitly specify what are the expected tags, schema keys and perhaps also endpoint paths?

```ts
expect(apiSpec.tags).to.eql([
  'CoffeeShops',
  'Users',
  // ...
]);

expect(Object.keys(apiSpec.components || {})).eql([
  'CoffeeShop',
  'User',
  // ...
]);
```

We already have tests to verify details of LB3 booter and Swagger-to-OpenAPI conversion, there is no need to repeat those detailed tests here again. IMO, it's important to test the high-level aspects of the generated spec:  Are CoffeeShop endpoints at expected URL paths? Are all model schemas included?